src/sage/quadratic_forms/quadratic_form__split_local_covering.py

```diff
@@ -254,7 +252,7 @@ def vectors_by_length(self, bound):
 
         # 3a. Increment (and carry if we go out of bounds)
         x[i] += 1
-        while (x[i] > L[i]) and (i < n-1):
+        while (x[i] > L[i]) and (i < n - 1):
```

Comment:
isn't it safer to first test that `i < n - 1` and then that `x[i] > L[i]` ? I'm not sure of the respective size of `x` and `L`.